### DISCUSSION TITLE
How do I grab a variable out of a script?

### DISCUSSION THREAD

**Discussion by BernardoGP4504** [CodenameCrew/CodenameEngine#258] [Q&A]
How do I do that on the title but with `Reflect.getProperty`? I need to know so I can make a FlxBar that tracks a script variable
I've tried `this`, `__script__` and even `scripts.getByName`

The issue is more specific as to the `o` field in `Reflect.getProperty`, as I can't seem to make getProperty recognize the script I want to grab a variable from

**Comment by NexIsDumb** [comment]
well you dont actually need the getProperty as every script has a map of normal, static, public variables (check the source to see them)

**Reply by BernardoGP4504** [reply]
But I need to know how to do it with getProperty, so I know how to declare the value variable on the `new FlxBar` function

**Reply by NexIsDumb** [reply]
from @NeeEoo :

var obj = {};

obj.health = 1;

obj, "health"
